capsules/src/net/udp/driver.rs

```diff
@@ -0,0 +1,494 @@
+//! UDP userspace interface for transmit and receive.
+//!
+//! Implements a userspace interface for sending and receiving UDP messages.
+//! Also exposes a list of interface addresses to the application (currently
+//! hard-coded.
+
+use core::cell::Cell;
+use core::{cmp, mem};
+use kernel::common::cells::TakeCell;
+use kernel::{AppId, AppSlice, Callback, Driver, Grant, ReturnCode, Shared};
+use net::ipv6::ip_utils::IPAddr;
+use net::udp::udp_recv::{UDPReceiver, UDPRecvClient};
+use net::udp::udp_send::{UDPSendClient, UDPSender};
+
+/// Syscall number
+pub const DRIVER_NUM: usize = 0x30002;
+
+const INTERFACES: [IPAddr; 2] = [
```

Comment:
What are these interfaces?

Reply:
They are intended to represent the interfaces available on the device. There are two for now because that simplified the process of writing a tx/rx test suite - one app uses interface 0, the other uses interface 1. Will be replaced soon with a reference to the source IP address associated with the 15.4 radio interface, once we assign a specific place in the kernel to hold that value.